src/main/java/uk/gov/hmcts/ccd/domain/service/caseaccess/CaseAccessOperation.java

```diff
@@ -50,4 +59,31 @@ public void revokeAccess(final String jurisdictionId, final String caseReference
                                  .collect(Collectors.toList());
     }
 
+    @Transactional
+    public void updateUserAccess(CaseDetails caseDetails, CaseUser caseUser) {
+        final Set<String> validCaseRoles = caseRoleRepository.getCaseRoles(caseDetails.getCaseTypeId());
+        final Set<String> targetCaseRoles = caseUser.getCaseRoles();
+
+        targetCaseRoles.stream()
```

Comment:
Maybe extraction to a self describing validation method would improve readability here?

Reply:
Indeed, very good point, done